### PR TITLE
refactoring: use hook replace LocaleReceiver

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactoring: use hook replace LocaleReceiver |
| 🇨🇳 Chinese | refactoring: 使用 useLocale 代替 LocaleReceiver 组件，删除 LocaleReceiver 组件 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/modal/PurePanel.tsx [hunk 1]
```diff
@@ -27,21 +28,24 @@ export function renderCloseIcon(prefixCls: string, closeIcon?: React.ReactNode)
...
-    onOk?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
-    onCancel?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
-  },
-) {
+interface FooterProps {
+  onOk?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  onCancel?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+}
+
+export const Footer: React.FC<
```
Comment (li-jia-nan) [inline components/modal/PurePanel.tsx:36]: 因为普通函数里面不能使用 hook，所以把 renderFooter 方法重构成了 FC

File: components/locale/LocaleReceiver.ts [hunk 2]
```diff
@@ -16,6 +16,8 @@ export interface LocaleReceiverProps<C extends LocaleComponentName = LocaleCompo
   ) => React.ReactElement;
 }
 
+// 这个方法没用了，用下面的 useLocale 代替，但是别删
+/* istanbul ignore next */
```
Comment (MadCcc) [inline components/locale/LocaleReceiver.ts:20]: 给个用例，加上 deprecated

Comment (MadCcc) [reply components/locale/LocaleReceiver.ts:20]: v4 好像文档里没提过了，直接删了吧

Comment (li-jia-nan) [reply components/locale/LocaleReceiver.ts]: done~

**Comment by MadCcc** [comment]
少了 200B，舒服了

**Review by MadCcc — APPROVED** [review]

**Comment by MadCcc** [comment]
Changelog 都写一下

**Comment by li-jia-nan** [comment]
为啥会少这么多，两个方法内部实现都基本一样

**Comment by li-jia-nan** [comment]
> Changelog 都写一下

done~

**Pull request merged**